File: tests/07_phases_ompss.dg/cxx/success_task_20.cpp
/*
<testinfo>
test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
</testinfo>
*/
#pragma omp task
void foo();

void bar()
{
    foo();
    #pragma omp taskwait
}

void foo()
{
    #pragma omp task
    {
    }
    #pragma omp taskwait
}

int main()
{
    foo();
}

File: tests/07_phases_ompss.dg/fortran/success_task_20.f90
! <testinfo>
! test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
! </testinfo>
SUBROUTINE FOO(X)
    INTEGER :: X
    X = X + 1
END SUBROUTINE FOO

PROGRAM P
    IMPLICIT NONE

    INTERFACE
        !$OMP TASK INOUT(X) FINAL(.TRUE.)
        SUBROUTINE FOO(X)
            INTEGER :: X
        END SUBROUTINE FOO
    END INTERFACE
    INTEGER X

    X = -1

     CALL FOO(X)
!$OMP TASKWAIT ON(X)
IF (X /= 0) THEN
    STOP -1
END IF

    !$OMP TASK INOUT(X) FINAL(.TRUE.)
         X = X + 1
     !$OMP END TASK
!$OMP TASKWAIT ON(X)

IF (X /= 1) THEN
    STOP -1
END IF

END PROGRAM P
